src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireTokenHandlerBase.cs

```diff
@@ -135,7 +135,16 @@ protected AcquireTokenHandlerBase(RequestData requestData)
 
                     this.NotifyBeforeAccessCache();
                     notifiedBeforeAccessCache = true;
-                    ResultEx = this.tokenCache.LoadFromCache(CacheQueryData, this.CallState);
+
+                    var aliasedAuthorities = await GetOrderedAliases(
+                        GetHost(this.Authenticator.Authority), this.Authenticator.ValidateAuthority, this.CallState).ConfigureAwait(false);
+                    foreach (var aliasedAuthority in aliasedAuthorities)
+                    {
+                        CacheQueryData.Authority = ReplaceHost(this.Authenticator.Authority, aliasedAuthority);
+                        ResultEx = this.tokenCache.LoadFromCache(CacheQueryData, this.CallState);
+                        if (ResultEx?.Result != null) break;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 control flow statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://iddp-sonarqube:9000/coding_rules#rule_key=csharpsquid%3AS134)